core/server/worker/src/main/java/alluxio/worker/netty/BlockReadHandler.java

```diff
@@ -91,7 +91,9 @@ protected void completeRequest(BlockReadRequestContext context) throws Exception
         }
       }
       if (!mWorker.unlockBlock(context.getRequest().getSessionId(), context.getRequest().getId())) {
-        mWorker.closeUfsBlock(context.getRequest().getSessionId(), context.getRequest().getId());
+        if (reader != null) {
+          mWorker.closeUfsBlock(context.getRequest().getSessionId(), context.getRequest().getId());
```

Comment:
Should we set reader to null after this? Maybe add a method to handle closing?

Reply:
done